sdks/java/core/src/main/java/org/apache/beam/sdk/io/CompressedSource.java

```diff
@@ -361,7 +281,7 @@ public void validate() {
   @Override
   protected final boolean isSplittable() throws Exception {
     return channelFactory instanceof FileNameBasedDecompressingChannelFactory
```

Comment:
<!--new_thread; commit:12876d43a961dd8ff9dba5f4cffcfbe643927eee; resolved:0-->
We should check to see if `channelFactory` is an instanceof `DecompressAccordingToFilename` otherwise if `FileNameBasedDecompressingChannelFactory` is made public, then `AUTO` may incorrectly say something is not compressed.

We should also detect that DoNotDecompress is the FileNamedBasedDecompressingChannelFactory and delegate to the sourceDelegate for isSplittable

`createSingleFileReader` should similarly be updated.

Reply:
Done - this was in fact buggy.